NitroxClient/GameLogic/InitialSync/PlayerPositionInitialSyncProcessor.cs

```diff
@@ -44,10 +47,22 @@ public override IEnumerator Process(InitialPlayerSync packet, WaitScreen.ManualW
             Player.main.ValidateEscapePod();
         }
 
+        Player.main.precursorOutOfWater = packet.InPrecursor;
+        Player.main.SetDisplaySurfaceWater(packet.DisplaySurfaceWater);
+
         Optional<NitroxId> subRootId = packet.PlayerSubRootId;
         if (!subRootId.HasValue)
         {
             yield return Terrain.WaitForWorldLoad();
+            Player.main.UpdateIsUnderwater();
+
+            // Check if Player might fall through the map
+            if (packet.InPrecursor || !Player.main.IsUnderwaterForSwimming())
+            {
+                // This coroutine must be started in a different thread to release the current one
```

Comment:
```suggestion
                // This coroutine must be started fresh to release the current one
```
Thread is maybe misleading